corehq/apps/registration/forms.py

```diff
@@ -594,7 +606,7 @@ def clean_email(self):
         email = self.cleaned_data['email'].strip()
 
         from corehq.apps.registration.validation import AdminInvitesUserFormValidator
-        error = AdminInvitesUserFormValidator.validate_email(self.domain, email)
+        error = AdminInvitesUserFormValidator.validate_email(self.domain, email, bool(self.invite))
```

Comment:
Does it make sense to skip all this validation if the field is set as read only? And then add a validation that the email didn't change like
```
    if self.fields['email'].widget.attrs.get('readonly', False):
        if email != self.invite.email:
            raise forms.ValidationError("The email field is read-only and cannot be changed.")
            ```

Reply:
I think what I have currently is simpler